.github/workflows/test.yaml

```
@@ -16,4 +28,10 @@ jobs:
           DOCKER_DEV_CI_MODE: 1
           DOCKER_BUILDKIT: 0
           DOCKER_QUIET_BUILD: 1
-        run: make -C utils/docker run DOCKER_DEV_OPTIONAL_CMD='env DOCKER_QUIET_BUILD=1 DOCKER_BUILDKIT=0 make all clobber'
\ No newline at end of file
+        run: |
+            DOCKER_BUILD_OPTS="--build-arg UBUNTU_VERSION=${{ matrix.os-version }} --build-arg UBUNTU_NAME=${{ matrix.os-name }}" \
+            # we pass again DOCKER_BUILD_OPTS since it's used in `make all` and not propagated by the initial `make run`
+            make -C utils/docker run DOCKER_DEV_OPTIONAL_CMD='env DOCKER_QUIET_BUILD=1 DOCKER_BUILDKIT=0 \
+                                                              DOCKER_BUILD_OPTS="--build-arg UBUNTU_VERSION=${{ matrix.os-version }} --build-arg UBUNTU_NAME=${{ matrix.os-name }}" \
+                                                              make all clobber
```

Slight rephrase and formatting tweak to point out symmetry of all three env vars 
```suggestion
            DOCKER_BUILD_OPTS="--build-arg UBUNTU_VERSION=${{ matrix.os-version }} --build-arg UBUNTU_NAME=${{ matrix.os-name }}" \
            make -C utils/docker run DOCKER_DEV_OPTIONAL_CMD='env DOCKER_QUIET_BUILD=1\
                                                                  DOCKER_BUILDKIT=0\
                                                                  DOCKER_BUILD_OPTS="--build-arg UBUNTU_VERSION=${{ matrix.os-version }} --build-arg UBUNTU_NAME=${{ matrix.os-name }}"\
                                                              make all clobber
            # we pass DOCKER_BUILD_OPTS, similar to DOCKER_QUITE_BUILD and DOCKER_BUILDKIT, to both 'make' invocations as 
            # they are required when building the containers as part of 'make run' and when building inside the container (as part of make all)
            # but make run does not automatically propagate environment variables to the container.
```